Feat: my-favorites

rkappes opened the pull request:
### Changes

Create the my-favorites page so users can see their saved favorite pets

### Comments
I initially implemented this just in the page.svelte and page.server.ts files in the /favorites. Then I changed it to use a component ```/components/Favorites.svelte```. I'm not sure which is the better or the 'correct' way to do it?
It successfully deletes the pet record from Supabase, but the user has to reload the page see the change in the UI. I wasn't sure how to implement this - I know svelte has the $state() rune, should I use this? or explicitly force a reload of the page?

### Work on next

- [ ] Add the insert favorite functionality

rkappes committed:
initial get favorites from supabase

rkappes committed:
initial display favorites

rkappes committed:
query user's favorties from favorites table

rkappes committed:
fix typo

rkappes committed:
retreive user's favorites from supabase favorite's table

rkappes committed:
display each pet favorite as a card

rkappes committed:
delete files in subfolder 'favorites'

rkappes committed:
Merge branch 'main' of https://github.com/planetaska/purrfect-match into my-favorites

rkappes committed:
move favorites into /favorites route

rkappes committed:
remove favorites from profile page

rkappes committed:
delete a favorite pet entry

rkappes committed:
remove actions as form not used for delete functionality

rkappes committed:
make new component 'Favorites' for displaying and deleting favorite pets

rkappes committed:
use Favorite component instead

rkappes committed:
remove extra lines

planetaska commented:
Never reload the page!!

With Svelte/SvelteKit pattern, what you should attempt to do is like this:
1. the data should almost always come from a load function, can be in the page.ts, server.page.ts, or layout.ts.
2. no matter where the load function is located, your goal is to pass data as a prop (aka: return the data).
3. then in your page, all you need to do is to use the $prop to receive data.
4. in your case, you want to delete some record and go back to the same page, so all you need to do is simply do a form post to the same page, delete the favorite inside the form action (using the id posted with the form - you may use a hidden field) to identify which favorite to delete), then simply return success. After the form is done, the load function will get run next, in turn get the updated status, in turn update the page (without reloading it!).

As rule of thumb, if you found yourself not following this simple pattern, and are doing something super complex for a simple feature, you are probably (or almost certainly) doing something wrong!

rkappes committed:
return to using a form to delete a pet record

rkappes committed:
remove having a favorites component

rkappes committed:
fix CSS so pet favorite cards are same size

rkappes committed:
Merge branch 'pets-layout' of https://github.com/planetaska/purrfect-match into my-favorites

rkappes committed:
add pet to favorites

rkappes committed:
use component AddFavorite instead

rkappes committed:
add petPhoto and URL

rkappes committed:
add retreiving pet photo

rkappes committed:
specify image height and width. Add link to pet record on PetFinder.com

rkappes committed:
add some css

rkappes committed:
Merge branch 'main' of https://github.com/planetaska/purrfect-match into my-favorites

rkappes commented:
Changes made:

- Used a form submission for the delete functionality
- added "add pet to favorites" functionality (option only visible if user is signed in)
   - added component AddFavorites
-  added some CSS for my favorites page

rkappes committed:
Merge branch 'main' of https://github.com/planetaska/purrfect-match into my-favorites
